frontend/src/features/public-form/utils/createSubmission.ts

```diff
@@ -78,20 +78,30 @@ export const createEncryptedSubmissionData = async ({
 }
 
 /**
+ * Used for both Email mode submissions and Storage mode submissions after encryption boundary shift.
  * @returns formData containing form responses and attachments.
  * @throws Error if form inputs are invalid.
  */
-export const createEmailSubmissionFormData = (
+export const createClearSubmissionFormData = (
```

Comment:
since version is actually non-optional for storage mode form, i feel that maybe a union type is better

`createClearSubmissionFormData = (formData: EmailSubmissionData | StorageSubmissionData)`

and then you can enforce that `StorageSubmissionData` contains `setBody: {version: number}`

Reply:
Thanks Shu Li! Good point, done in [9847df0](https://github.com/opengovsg/FormSG/pull/6665/commits/9847df0b2b21db1c5238a549e41718f7cbf0d03d)